File: packages/android/src/main/java/com/kozmos/components/Checkbox/Checkbox.figma.kt
package com.kozmos.components.checkbox

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=77-1410")
class KozmosCheckboxConnect {
    @FigmaProperty(FigmaType.Enum, "Checked")
    val checked: Boolean = Figma.mapping(
        "Unchecked" to false,
        "Checked" to true
    )

    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Accept terms"

    @FigmaProperty(FigmaType.Enum, "State")
    val state: String = Figma.mapping(
        "Default" to "default",
        "Disabled" to "disabled",
        "Error" to "error"
    )

    @Composable
    fun ComponentExample() {
        KozmosCheckbox(
            checked = checked,
            onCheckedChange = {},
            label = label,
            enabled = state != "disabled",
            error = state == "error"
        )
    }
}

File: packages/ios/Sources/Components/Checkbox/Checkbox.figma.swift
import SwiftUI
import Figma

struct KozmosCheckboxConnect: FigmaConnect {
    let component = KozmosCheckbox.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=77-1410"

    @FigmaEnum(
        "Checked",
        mapping: [
            "Unchecked": false,
            "Checked": true
        ]
    )
    var checked: Bool = false

    @FigmaString("Label Text")
    var label: String = "Accept terms"

    @FigmaEnum(
        "State",
        mapping: [
            "Default": "default",
            "Disabled": "disabled",
            "Error": "error"
        ]
    )
    var state: String = "default"

    var body: some View {
        KozmosCheckbox(
            checked: .constant(self.checked),
            label: self.label,
            disabled: self.state == "disabled",
            error: self.state == "error"
        )
    }
}
